### PR TITLE
Apply ignore area to all images from selected ignore area from details page.

### PR DESCRIPTION
**Pull request by suratdas**
https://github.com/Visual-Regression-Tracker/Visual-Regression-Tracker/issues/276 
Attached video. Will raise backend pull request if this is approved. Currently, it is not possible to select which test runs it needs to apply to.

https://user-images.githubusercontent.com/9042580/122833949-9e75fe80-d2a2-11eb-9116-be2fa6651b89.mov

### CODE REVIEW

**Comment by pashidlos** [comment]
@suratdas sorry for delay
I changed API to support bulk operations 
not you could utilise `POST test-runs/ignoreAreas/add`

**Comment by suratdas** [comment]
> @suratdas sorry for delay
> I changed API to support bulk operations
> not you could utilise `POST test-runs/ignoreAreas/add`

I have included the latest changes now.

File: src/components/TestDetailsModal.tsx [hunk 1]
```diff
@@ -151,6 +157,11 @@ const TestDetailsModal: React.FunctionComponent<{
     }
   };
 
+  const getIgnoreAreaForSelectionId = (idPassed: string) => {
+    let ignoreAreaToReturn = ignoreAreas.filter((area) => idPassed === area.id)[0];
+    return ignoreAreaToReturn;
```
Comment (pashidlos) [inline src/components/TestDetailsModal.tsx:162]: could be replaced with `find` as only one is needed 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

File: src/components/TestDetailsModal.tsx [hunk 2]
```diff
@@ -180,6 +191,24 @@ const TestDetailsModal: React.FunctionComponent<{
     setStageOffset(defaultStagePos);
   };
 
+  const applyIgnoreArea = () => {
+    setProcessing(true);
+    let newIgnoreArea = getIgnoreAreaForSelectionId(selectedRectId!);
+    testRunService.getList(testRun.buildId).then(
+      (allRows) => {
+        let allIds: string[] = [];
+        allRows.forEach((value) => allIds.push(value.id));
```
Comment (pashidlos) [inline src/components/TestDetailsModal.tsx:200]: you could make it with `map`
const allIds = allRows.map((item) => item.id);

File: src/components/TestDetailsModal.tsx [hunk 3]
```diff
@@ -180,6 +191,24 @@ const TestDetailsModal: React.FunctionComponent<{
     setStageOffset(defaultStagePos);
   };
 
+  const applyIgnoreArea = () => {
+    setProcessing(true);
+    let newIgnoreArea = getIgnoreAreaForSelectionId(selectedRectId!);
+    testRunService.getList(testRun.buildId).then(
+      (allRows) => {
```
Comment (pashidlos) [inline src/components/TestDetailsModal.tsx:198]:
```suggestion
      (testRuns: TestRun[]) => {
```

File: src/components/TestDetailsModal.tsx (hunk 1 above)
Comment (pashidlos) [reply src/components/TestDetailsModal.tsx:162]: also handle the case if it's `undefined`

File: src/components/TestDetailsModal.tsx [hunk 4]
```diff
@@ -180,6 +191,24 @@ const TestDetailsModal: React.FunctionComponent<{
...
+        let allIds: string[] = [];
+        allRows.forEach((value) => allIds.push(value.id));
+        let data: UpdateIgnoreAreaDto = { ids: allIds, ignoreAreas: [newIgnoreArea] };
+        testRunService.addIgnoreAreas(data).then(() => {
```
Comment (pashidlos) [inline src/components/TestDetailsModal.tsx:202]: add error case handling

**Review by pashidlos — CHANGES_REQUESTED** [review]

**Review by pashidlos — APPROVED** [review]
LGTM

**Comment by pashidlos** [comment]
@suratdas thanks for contribution!

**Pull request merged**